src/Symfony/Component/Form/Console/Descriptor/TextDescriptor.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Form\Console\Descriptor;
+
+use Symfony\Component\Console\Helper\TableSeparator;
+use Symfony\Component\Form\ResolvedFormTypeInterface;
+
+/**
+ * @author Yonel Ceruto <yonelceruto@gmail.com>
+ *
+ * @internal
+ */
+class TextDescriptor extends Descriptor
+{
+    protected function describeResolvedFormType(ResolvedFormTypeInterface $resolvedFormType, array $options = array())
+    {
+        $this->collectOptions($resolvedFormType);
+
+        $options = $this->normalizeAndSortOptionsColumns(array_filter(array(
+            'own' => $this->ownOptions,
+            'overridden' => $this->overriddenOptions,
+            'parent' => $this->parentOptions,
+            'extension' => $this->extensionOptions,
+        )));
+
+        // setting headers and column order
+        $tableHeaders = array_intersect_key(array(
+            'own' => 'Options',
+            'overridden' => 'Overridden options',
+            'parent' => 'Parent options',
+            'extension' => 'Extension options',
+        ), $options);
+
+        $tableRows = array();
+        $count = count(max($options));
+        for ($i = 0; $i < $count; ++$i) {
+            $cells = array();
+            foreach (array_keys($tableHeaders) as $group) {
+                if (isset($options[$group][$i])) {
+                    $option = $options[$group][$i];
+
+                    if (is_string($option) && $this->isRequiredOption($option)) {
+                        $option .= ' <info>(required)</info>';
+                    }
+
+                    $cells[] = $option;
+                } else {
+                    $cells[] = null;
+                }
+            }
+            $tableRows[] = $cells;
+        }
+
+        $this->output->title(sprintf('%s (Block prefix: "%s")', get_class($resolvedFormType->getInnerType()), $resolvedFormType->getInnerType()->getBlockPrefix()));
+        $this->output->table($tableHeaders, $tableRows);
+
+        if ($this->hasParents()) {
```

Comment:
`if ($parents = $this->getParents())` ? extra internal api is imo. unneeded. Or just use the property direclty as it's protected.